lib/public-stats.js

```diff
@@ -282,15 +284,84 @@ const updateRetrievalTimings = async (pgClient, committees) => {
  * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealClients
  */
 export const updateDailyClientRetrievalStats = async (pgClient, committees, findDealClients) => {
-  const retrievalStatsPerClient = new Map()
+  const flatStats = buildPerPartyStats(committees, findDealClients, 'client')
+
+  await pgClient.query(`
+    INSERT INTO daily_client_retrieval_stats (
+      day,
+      client_id,
+      total,
+      successful,
+      successful_http
+    ) VALUES (
+      now(),
+      unnest($1::text[]),
+      unnest($2::int[]),
+      unnest($3::int[]),
+      unnest($4::int[])
+    )
+    ON CONFLICT (day, client_id) DO UPDATE SET
+      total = daily_client_retrieval_stats.total + EXCLUDED.total,
+      successful = daily_client_retrieval_stats.successful + EXCLUDED.successful,
+      successful_http = daily_client_retrieval_stats.successful_http + EXCLUDED.successful_http
+  `, [
+    flatStats.map(stat => stat.partyId),
+    flatStats.map(stat => stat.retrievalStats.total),
+    flatStats.map(stat => stat.retrievalStats.successful),
+    flatStats.map(stat => stat.retrievalStats.successfulHttp)
+  ])
+}
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealAllocators
+ */
+export const updateDailyAllocatorRetrievalStats = async (pgClient, committees, findDealAllocators) => {
+  const flatStats = buildPerPartyStats(committees, findDealAllocators, 'allocator')
+
+  await pgClient.query(`
+    INSERT INTO daily_allocator_retrieval_stats (
+      day,
+      allocator_id,
+      total,
+      successful,
+      successful_http
+    ) VALUES (
+      now(),
+      unnest($1::text[]),
+      unnest($2::int[]),
+      unnest($3::int[]),
+      unnest($4::int[])
+    )
+    ON CONFLICT (day, allocator_id) DO UPDATE SET
+      total = daily_allocator_retrieval_stats.total + EXCLUDED.total,
+      successful = daily_allocator_retrieval_stats.successful + EXCLUDED.successful,
+      successful_http = daily_allocator_retrieval_stats.successful_http + EXCLUDED.successful_http
+  `, [
+    flatStats.map(stat => stat.partyId),
+    flatStats.map(stat => stat.retrievalStats.total),
+    flatStats.map(stat => stat.retrievalStats.successful),
+    flatStats.map(stat => stat.retrievalStats.successfulHttp)
+  ])
+}
+/**
+* @param {Iterable<Committee>} committees
+* @param {(minerId: string, cid: string) => (string[] | undefined)} findDealActors
+* @param {string} partyName
```

Comment:
```suggestion
* @param {'client' | 'allocator'} partyName
```

We could improve typing here by using string literal types instead of `string`.